ctest.h

```diff
@@ -16,6 +16,10 @@
 #ifndef CTEST_H
 #define CTEST_H
 
+#if !(defined (_GNU_SOURCE) || defined (_BSD_SOURCE) || defined(_POSIX_C_SOURCE))
+#undef CTEST_SEGFAULT
```

Comment:
On my list, Linux, Bsd and Mac are all target systems. Windows comes trailing behind by a mile or so. Nice if we can support with with minor changes, but I don't want to have to go to great lengths for it..